### PR TITLE
Add an error page instead of redirecting to 404 when the redirect uri param is wrong

### PR DESCRIPTION
**Pull request by gregory-latinier**

### CODE REVIEW

**Comment by bonustrack** [comment]
Can we create an API (post) route that check for the `redirect_uri` instead of check directly on server? We can create a route `/api/apps/@:clientId/verify`. This way we can also remove the route `/oauth2/authorize`.

File: src/routes.js [hunk 1]
```diff
@@ -54,6 +55,7 @@ export default (
     <Route path="/authorize/@:username" component={SignAuthorize} />
     <Route path="/authorize/@:username/:role" component={SignAuthorize} />
     <Route path="/revoke/@:username" component={Revoke} />
+    <Route path="/error" component={Error} />
```
Comment (bonustrack) [inline src/routes.js]: Is this still used?

File: routes/apps.js [hunk 2]
```diff
@@ -143,4 +143,23 @@ router.all('/authorized', authenticate('user'), async (req, res) => {
   res.json({ apps });
 });
 
+router.post('/@:clientId/verify', async (req, res) => {
+  const { clientId } = req.params;
+  const { redirectUri } = req.body;
```
Comment (bonustrack) [inline routes/apps.js]: It would be better if we keep the same casing for all query parameters across apps which is snake_case and not camelCase

**Comment by bonustrack** [comment]
I just noticed that the error components block does not have minimum width:
image
https://sc2-master-pr-269.herokuapp.com/oauth2/authorize?client_id=busy4pp&redirect_uri=xxx
Also font-size of the error is too small. Can we use the same error component with same style for both authorize page and hot signing? Like this:
https://steemconnect.com/sign/vote?author=krnel&permlink=getting-tdal-with-ai-assistants&weight=10000&auto_return=true&redirect_uri=https://google.fr

**Comment by bonustrack** [comment]
Assuming this is complete, is there a reason to not using same error component than hot signing?

**Comment by gregory-latinier** [comment]
The error in hot sign is not a standalone component.

File: src/components/Error.less [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.Error_frame {
+  min-width: 300px;
```
Comment (bonustrack) [inline src/components/Error.less]: This is still too small, style should be `max-width: 500px; width: 100%` so it not fixed width on mobile. Actually the class `Sign_frame` has the proper style already, we could rely on the same style class we dont need to create Error.less file. You just need to figure out why `Sign_frame` width styles is not working on this page
image

File: src/components/Error.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import React, { PropTypes } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { Link } from 'react-router';
+
+const Error = ({ error }) =>
+  <div className="Sign">
+    <div className="Sign_frame Error_frame">
```
Comment (bonustrack) [inline src/components/Error.js]: You should remove `Error_frame` here

File: src/components/Error.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import React, { PropTypes } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { Link } from 'react-router';
+
+const Error = ({ error }) =>
+  <div className="Sign">
```
Comment (bonustrack) [inline src/components/Error.js]: The style is still not working properly, here there should be `className="Sign_content"`, Sign class is set on the parent component already. Perhaps we will also need the style `Sign.less` imported on the parent component `Authorize.js`

**Pull request closed without merging**